### PR TITLE
Load admin monitors popup content via ajax

### PR DESCRIPTION
**Pull request by fqueiruga**
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

This PR aims to make loading the list of administrative monitors asynchronous and to not affect page load. This also has the benefit of fixing keyboard navigation through the page header.

No notable changes to show in screenshots, except the empty state 

<details>
<Summary>Screenshot of empty state</summary>

This should almost never be shown (for example, user gets permissions to read monitors removed and tries to load them before reloading the page).

Captura de pantalla 2020-11-19 a las 17 09 21

</details>

### Proposed changelog entries

* Load the administrative monitors popup on demand to reduce page load time.
* Fixes a bug where the page header would not be navigable using keyboard when there were active administrative monitors.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@timja
@daniel-beck 
@Wadeck 
@oleg-nenashev

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by timja** [comment]
> Fixes a bug where the page header would not be navigable using keyboard when there were active administrative monitors.

nice! (if you keep tabbing you can navigate but there's some invisible elements to get through first)

**Comment by fqueiruga** [comment]
Exactly that one

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.jelly
Comment (timja) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.jelly:72]: looks like the indent change should be reverted here?

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.js
Comment (timja) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.js:19]: would be good to extract this to a variable so that people who don't know keycodes know what this is

**Review by timja — COMMENTED** [review]
looks really good, only minor comments

**Comment by timja** [comment]
> Exactly that one, shouldn't happen anymore

yes, tested that, and it works now

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsApiData/monitorsList.properties
Comment (timja) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsApiData/monitorsList.properties:1]: what about this?
```suggestion
no_active_monitors=There are currently no active warnings
```

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.jelly
Comment (fqueiruga) [reply core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.jelly]: YES

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.js
Comment (fqueiruga) [reply core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.js]: I put a comment next to it to clarify, if that's OK.

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.js
Comment (timja) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.js:21]: I meant something like this, what do you think?
```suggestion
            var escapeKeyCode = 27;
            if (e.keyCode === escapeKeyCode) {
```

**Review by timja — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by fqueiruga** [comment]
Seems that the ATH tests are being aborted. Any idea what may be happening?

**Review by fcojfernandez — APPROVED** [review]
At some point the administrative monitors should be moved to the header from the footer IMO, but as a first step it's great

File: core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java:23]: Should these calls have explicit permission checks?

Comment (fqueiruga) [reply core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java:23]: The authorization is implicit through the decorator. That said I think it would make sense to have more explicit auth. I have no idea how to got about this. If anyone wants to commit straight to my branch please go ahead :D

**Comment by fqueiruga** [comment]
Any chance this could make it this week? :D

File: core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java
Comment (fqueiruga) [reply core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java:23]: @daniel-beck @Wadeck  how would you go about this?

Comment (timja) [reply core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java:23]: it's quite common to rely on permission checks in this way as far as I know

**Review by oleg-nenashev — APPROVED** [review]
Should be good to go.

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

File: core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java
Comment (Wadeck) [reply core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java:23]: In this case it's a bit special because the permission is checked deeper in the code. It was recently changed from Admin to SystemRead. It means that if we have an additional check here, it's a bit a duplicate code, so I am fine with the current approach.

Of course if the method is starting to do something else, the explicit permission will be required, but we should not be future-proof without good rational.

**Comment by Wadeck** [comment]
Please do not merge, I am working on some quick changes here to prevent "regression" from the admin monitor separation in terms of performance.

**Comment by Wadeck** [comment]
👍 really nice job @fqueiruga :)

I adjusted a bit the code and added non-regression tests. I did not expect someone to remove that "feature" for the performance aspect, so now it's covered ;) In your PR the page was asking 3x the getMonitorList.

After some PRs like this one you will no longer be able to say you're not a Java developer ;)

**Review by Wadeck — APPROVED** [review]
Test added + rollback the filtering removal to keep the 1x call instead of 3x

File: core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java
Comment (fqueiruga) [inline core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java]: why aren't this needed?

Comment (Wadeck) [reply core/src/main/java/jenkins/management/AdministrativeMonitorsApi.java]: It's done automatically by the JellyFacet (need to find exactly the code, inside Stapler, digging...)

**Review by timja — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
Looks good. Just in case I checked improper usages of the restricted APIs, all looks good. 🚢 🇮🇹

**Comment by daniel-beck** [comment]
This PR causes:

```
WARNING	j.s.s.StaplerFilteredActionListener#onFieldTrigger: New Stapler routing rules result in the URL "/administrativeMonitors/securityPopupContent" no longer being allowed. If you consider it safe to use, add the following to the whitelist: "field jenkins.model.Jenkins administrativeMonitors". Learn more: https://jenkins.io/redirect/stapler-routing
```

Due to the naming collision between the field in Jenkins https://github.com/jenkinsci/jenkins/blob/9a8588951b77a938fcf4ba9cee68d4c3e98d28b0/core/src/main/java/jenkins/model/Jenkins.java#L722 and the `getDynamic` fallback to `RootAction` https://github.com/jenkinsci/jenkins/blob/9a8588951b77a938fcf4ba9cee68d4c3e98d28b0/core/src/main/java/jenkins/model/Jenkins.java#L3787

**Comment by daniel-beck** [comment]
Probably caused [JENKINS-70036](https://issues.jenkins.io/browse/JENKINS-70036).

**Pull request merged**
